static/js/camera.js

```diff
@@ -0,0 +1,43 @@
+window.onload = function() {
+  updateSnapshotLoop()
+};
+
+var snapshotCount = 0;
+
+function restartCameraViewing() {
+  document.getElementById("snapshot-stopped").style.display = 'none';
+  document.getElementById("snapshot-image").style.display = '';
+  snapshotCount = 0;
+  updateSnapshotLoop();
+}
+
+async function updateSnapshotLoop() {
+  var xmlHttp = new XMLHttpRequest();
+  xmlHttp.open("PUT", "/api/camera/snapshot", true);
+  xmlHttp.onreadystatechange = async function() {
+    if (xmlHttp.readyState == 4 && xmlHttp.status == 200) {
+      let snapshot = document.getElementById("snapshot-image")
+      snapshot.src = "/camera/snapshot?"+ new Date().getTime();
+      await sleep(500);
+      snapshotCount++;
+      if (snapshotCount < 200) {
+        updateSnapshotLoop();
+      } else {
+        stopSnapshots('Timeout for camera viewing.');
+      }
+    } else if (xmlHttp.readyState == 4 && xmlHttp.status != 200) {
+      stopSnapshots('Error getting new snapshot.')
+    }
+  }
```

Comment:
Instead of using `onreadystatechange` why not use `onload` and `onerror`? You don't need to worry about checking `readyState` then.  You still need to check `status` in the onload handler though.

See also: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/Using_XMLHttpRequest

Reply:
Also, if there's an error it might be good to report the HTTP status code, either to the UI or the console log.